index.js

```diff
@@ -4,6 +4,9 @@ import getRawBody from 'raw-body'
 import assert from 'http-assert'
 import { validate } from './lib/validate.js'
 
+// Approximate length of each SPARK/Meridian round in milliseconds
+const ROUND_LENGTH_IN_MS = 30_000 /* ms */
```

Comment:
@juliangruber Is there a way to read this information from the contract? Is it the `nextRoundLength` variable?

Do you think it's worth the effort to remove this constant and query the round length whenever we detect the start of a new MERidian round?

Considering how infrequently we are going to change this constant, maybe it's fine to keep this hard-coded for now.

Reply:
BTW, the smart contract code says that the next round length is 10:

https://github.com/Meridian-IE/impact-evaluator/blob/9242c15bfeafcc89a23e91caa34532b4bd09850a/src/ImpactEvaluator.sol#L17
```sol
    uint public nextRoundLength = 10;
```

Reply:
> @juliangruber Is there a way to read this information from the contract? Is it the nextRoundLength variable?

Yes! Any public variable has a getter automatically. You can get it via `ie.nextRoundLength()`

> Do you think it's worth the effort to remove this constant and query the round length whenever we detect the start of a new MERidian round?
Considering how infrequently we are going to change this constant, maybe it's fine to keep this hard-coded for now.

We could either query at the start of each round, or at process start? It should be fine to query at the start of each round, since the call is free and should be reasonably fast.

> BTW, the smart contract code says that the next round length is 10 (~5min?)

Yes, this is the default state. I've since changed the round length via `ie.setNextRoundLength(uint _nextRoundLength)`

Reply:
> Yes, this is the default state. I've since changed the round length via `ie.setNextRoundLength(uint _nextRoundLength)`

Yeah, I remember. It makes me wonder though, if we will remember to bump it up whenever we deploy a new version of the smart contract 🤔

Reply:
I wondered about this too. A spark deployment script would be great, which is independent of the meridian deployment script

Reply:
However, the manual process works for me still, since
1. We expect to deploy the contract only once
2. It's not hard to course correct